webapp/components/compat-2021.js

```diff
@@ -461,124 +623,135 @@ class Compat2021FeatureChart extends PolymerElement {
+    // this only exposes browser changes and not test suite changes.
+    const browserVersions = await this.dataManager.getBrowserVersions(stable);
     chart.setAction({
-      id: 'revisionDiff',
-      text: 'Show diff from previous release',
+      id: 'revisionChangelog',
+      text: 'Show browser changelog',
       action: () => {
         let selection = chart.getSelection();
         let row = selection[0].row;
         let column = selection[0].column;
 
-        // Not implemented for Firefox or Safari yet.
-        if (column !== 1) {
-          alert('Diff only supported for Chrome currently');
-          return;
-        }
-
         // Map from the selected column to the browser index. In the datatable
         // Chrome is 1, Firefox is 3, Safari is 5 => these must map to [0, 1, 2].
-        let browserIdx = (column - 1) / 2;
+        let browserIdx = Math.floor(column / 2);
 
         let version = browserVersions[browserIdx][row];
         let lastVersion = version;
         while (row > 0 && lastVersion === version) {
           row -= 1;
           lastVersion = browserVersions[browserIdx][row];
         }
-        // TODO: If row == -1, we've failed, but we should grey out the
-        // option instead in that case.
-        window.open(this.getChromeDiffUrl(lastVersion, version));
+        // TODO: If row == -1 here then we've failed.
+
+        if (browserIdx === 0) {
+          window.open(this.getChromeChangelogUrl(lastVersion, version));
+          return;
+        }
+
+        if (browserIdx === 1) {
+          if (stable) {
+            window.open(this.getFirefoxStableChangelogUrl(lastVersion, version));
+          } else {
+            this.$.firefoxNightlyDialog.open();
+          }
+          return;
+        }
+
+        this.$.safariDialog.open();
       },
     });
 
-    chart.draw(dataTable, this.chartOptions);
+    chart.draw(dataTable, this.getChartOptions(feature));
   }
 
-  getChromeDiffUrl(fromVersion, toVersion) {
+  getChromeChangelogUrl(fromVersion, toVersion) {
     // Strip off the 'dev' suffix if there.
     fromVersion = fromVersion.split(' ')[0];
     toVersion = toVersion.split(' ')[0];
     return `https://chromium.googlesource.com/chromium/src/+log/${fromVersion}..${toVersion}?pretty=fuller&n=10000`;
   }
 
-  createTooltip(browser, version, score) {
-    return `${browser} ${version}: ${score.toFixed(3)}`;
+  getFirefoxStableChangelogUrl(fromVersion, toVersion) {
+    // The version numbers are reported as XX.Y.Z, but pushlog wants
+    // 'FIREFOX_XX_Y_Z_RELEASE'.
+    const fromParts = fromVersion.split('.');
+    const fromRelease = `FIREFOX_${fromParts.join('_')}_RELEASE`;
+    const toParts = toVersion.split('.');
+    const toRelease = `FIREFOX_${toParts.join('_')}_RELEASE`;
+    return `https://hg.mozilla.org/mozilla-unified/pushloghtml?fromchange=${fromRelease}&tochange=${toRelease}`;
+  }
+
+  onFirefoxNightlyDialogGo() {
```

Comment:
`onFirefoxNightlyDialogGo` look like pretty unusual naming to me. `openFirefoxNightlyDialog`? Or what does the "go" bit mean?

Reply:
Oh, the 'Go' button is in the pop-up dialog and this is the handler for it. I'll add the word `Button` or something to clarify a bit.